docs/conf.py

```diff
@@ -50,8 +48,9 @@
 # This pattern also affects html_static_path and html_extra_path.
 exclude_patterns = ['_build']
 
-# Sphinx 4.0+ required for autodoc_typehints_description_traget
-needs_sphinx = '4.0'
+# Sphinx 5.3+ required to allow section titles inside autodoc class docstrings
+# https://github.com/sphinx-doc/sphinx/pull/10887
```

Comment:
I would also be okay with removing section headings from docstrings, I don't imagine it happens in very many places.